### PR TITLE
Added new changes

### PR DESCRIPTION
**Pull request by jacknormand**
This pull request contains my changes. Once reviewed and approved, I will create a pull request for my fork.

What I changed: I added a new text input field for "Max number of sensors". This allows the user to put a cap on how many sensors they retrieve data from. It is fully integrated with the code and is operational.

### CODE REVIEW

File: purple_air.js [hunk 1]
```diff
@@ -433,20 +444,39 @@ var purple_air = {
         let fetch_purple_air = await (await fetch(BASE_PURPLE_AIR_URL)).json()
 
         let data = fetch_purple_air.data
-        for (let d of data) {
```
Comment (deekshitharajoli) [inline purple_air.js:436]: This new if else is adding redundant code as the same code is copied in both places. Consider below alternative solution below.  We can keep track of the number of sensors data reading and break the loop if max cap is reached.  Below solution doesn't duplicate the code. 

```
let totalReqSensors = purple_air.state.sensorNum || 0
let recordsCount = 0;

for (let d of data) {
    let newRow = { ...purple_air_fields }
    newRow.Location = purple_air.state.city
    newRow.sensor_index = d[0]
    newRow.name = d[1]
    // newRow.primary_id_a =   d[2]
    // newRow.primary_key_a =  d[3]
    newRow.latitude = d[2]
    newRow.longitude = d[3]
    // latLngList.push( `${d[2]},${d[3]}` )
    sensorValues.push(newRow)

    // Keep count of the sensor records read and terminate if max cap is reached
    recordsCount = recordsCount + 1; 
    if(totalReqSensors > 0 && recordsCount >= totalReqSensors){
    	break;
    }
}

```

**Review by deekshitharajoli — CHANGES_REQUESTED** [review]

File: purple_air.js (hunk 1 above)
Comment (jacknormand) [reply purple_air.js:436]: Thank you!

**Comment by igorsteinmacher** [comment]
@jacknormand let me know when the new code is available so I can take a look

**Comment by jacknormand** [comment]
I added the changes to my fork. They are on my main branch and I closed the pull request so here <https://github.com/jacknormand/codappurpleairplugin/commit/d6850faddf2a4bdec765e4add8c211e486c8cf69> are the changes. Let me know if there is anything else I need to do.

Thanks, Jack

**Comment by deekshitharajoli** [comment]
Reviewed the updated code changes, its working fine in the instance.

File: index.html [hunk 2]
```diff
@@ -156,6 +156,22 @@
           </select>
     </div>
 
+    <div class="input-group mt-2">        
+        <i class="input-group-text bi bi-123"></i>
```
Comment (SaiSriMuttineni) [inline index.html:160]: Dear Jack,

The new field added has the field name displayed as "123". Please change the code accordingly to display the field name as something meaningful such as the Number of Sensors to be Detected or anything appropriate.

Also, the placeholder text is too long(an empty field would be fine). I think this text could be added as an info tab.

Thanks
SaiSri.

**Pull request closed without merging**